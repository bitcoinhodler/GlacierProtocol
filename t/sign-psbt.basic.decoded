{
  "txid": "99207d2ce53cf1c8aedab5df8e28fcba90089015fdf4e033323064e2b7921a47",
  "hash": "2d491941e214f6e3695b92eadb7f038e2c8aaac6f5079c8a5adacf8577993e8b",
  "version": 2,
  "size": 406,
  "vsize": 190,
  "weight": 760,
  "locktime": 0,
  "vin": [
    {
      "txid": "e08c08528e9984a529c26ad82ba3521e06bf602b0cf6b535bdc3c9a2c2d8a201",
      "vout": 1,
      "scriptSig": {
        "asm": "0020dabea3445c14e4a08d6705db4373bef467d4c64e7c8ddf149be50670de6878ae",
        "hex": "220020dabea3445c14e4a08d6705db4373bef467d4c64e7c8ddf149be50670de6878ae"
      },
      "txinwitness": [
        "",
        "30440220127ee252293140abf65891e22aa4a95814c00654ded5705a5ad0640001392ad402202ecbaf243f217ff4720763b14dd33e65d4d953fa8c857edbd707ae965a2a745401",
        "30440220206fdc880caa88c8378b2fd95baa112947836dd4983abfbe51357133ed0856ca02201dc760c8c08716c92ef969dc511b72df057f37d434b84b8dcd4da7b240aee08c01",
        "5221029f531503facdac2496f50a446d9bd29846a06a04a45e3845b656bb471df422fc2102e30787703a990e4015a2cb9071fcfd1c7d4641fb294e4b4c3f5f6b450a1925132102da28088a8022651171c4f13429b98709dabe13bc6da526537fdd2d0730dd2dbb2103286c96ecaa850a6ba43cc45fbb539c1fb1d65c23cc0f3cd09fcf9765826ff9de54ae"
      ],
      "sequence": 4294967293
    }
  ],
  "vout": [
    {
      "value": 0.17399809,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 d780cea68b8bc5b8c3c375c040c2cce08c9a9077 OP_EQUAL",
        "hex": "a914d780cea68b8bc5b8c3c375c040c2cce08c9a907787",
        "reqSigs": 1,
        "type": "scripthash",
        "addresses": [
          "2NCthaDAsJUJ2q5s1L4HhexUGMJ5t16vqer"
        ]
      }
    }
  ]
}
