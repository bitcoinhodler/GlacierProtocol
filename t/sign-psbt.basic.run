#!/bin/bash
set -e

$GLACIERSCRIPT --regtest=$1 sign-psbt << INPUT
INPUT
