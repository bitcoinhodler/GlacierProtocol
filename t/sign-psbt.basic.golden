Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?Input a filename located in the current directory which contains the PSBT:
0.17400000 BTC in unspent inputs from cold storage address 2MzqiaZzpLT2SSBfsFqqo3FpZsP8g6WTvyC
0.17399809 BTC going to destination address 2NCthaDAsJUJ2q5s1L4HhexUGMJ5t16vqer
Fee amount: 0.00000191
Confirm? (y/n): 
How many private keys will you be signing this transaction with? 
#: Key #1: Key #2: 
Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.17400000 BTC in unspent inputs from cold storage address 2MzqiaZzpLT2SSBfsFqqo3FpZsP8g6WTvyC
0.17399809 BTC going to destination address 2NCthaDAsJUJ2q5s1L4HhexUGMJ5t16vqer
Fee amount: 0.00000191

Signing with private keys: 
cVb7UNXC7nXzBLEtQZMzJcoRYNsiGCTjgAiibJqtEpLM1gimrhW2
cUxC3su6U61NsGspvcsD7JxSDUiRY7YaybTHGDgF5e8T5tpk7UD1


Confirm? (y/n): 
Calculating transaction...


Raw signed transaction (hex):
0200000000010101a2d8c2a2c9c3bd35b5f60c2b60bf061e52a32bd86ac229a584998e52088ce00100000023220020dabea3445c14e4a08d6705db4373bef467d4c64e7c8ddf149be50670de6878aefdffffff01018009010000000017a914d780cea68b8bc5b8c3c375c040c2cce08c9a90778704004730440220127ee252293140abf65891e22aa4a95814c00654ded5705a5ad0640001392ad402202ecbaf243f217ff4720763b14dd33e65d4d953fa8c857edbd707ae965a2a7454014730440220206fdc880caa88c8378b2fd95baa112947836dd4983abfbe51357133ed0856ca02201dc760c8c08716c92ef969dc511b72df057f37d434b84b8dcd4da7b240aee08c018b5221029f531503facdac2496f50a446d9bd29846a06a04a45e3845b656bb471df422fc2102e30787703a990e4015a2cb9071fcfd1c7d4641fb294e4b4c3f5f6b450a1925132102da28088a8022651171c4f13429b98709dabe13bc6da526537fdd2d0730dd2dbb2103286c96ecaa850a6ba43cc45fbb539c1fb1d65c23cc0f3cd09fcf9765826ff9de54ae00000000

Transaction fingerprint (md5):
eed333304281d40c1fec8513e95a9b57
QR code for transaction written to transaction.png
